src/zinoargus/__init__.py

```diff
@@ -237,68 +238,80 @@ def synchronize_continuously(argus_incidents: IncidentMap, zino_cases: CaseMap):
             continue
         _logger.debug(
             "Update on Zino case id:%s type:%s info:%s",
             update.id,
             update.type,
             update.info,
         )
+
+        # Ensure we have the details on both the Zino Case and Argus Incident being updated
+        if update.id not in zino_cases:
+            # We didn't know about this case ID before, so we need to fetch it
+            zino_cases[update.id] = _zino.case(update.id)
+        case = zino_cases[update.id]
+
+        if not is_case_interesting(case):
+            # Ignore this update, as it's not interesting to us
+            continue
+
+        incident = get_or_make_argus_incident_for_zino_case(
+            update.id, case, argus_incidents
+        )
+
         if update.type == "state":
-            old_state, new_state = update.info.split(" ", 1)
-            if new_state == "closed":
-                # Closing case
-                _logger.debug(
-                    "Zino case %s is closed and is being removed from argus", update.id
-                )
-                if update.id in argus_incidents:
-                    close_argus_incident(
-                        argus_incidents[update.id],
-                        description="Zino case closed by user",
-                    )
-                    del argus_incidents[update.id]
-                else:
-                    _logger.info(
-                        "Can't close zino case %s because it's not found in argus",
-                        update.id,
-                    )
-
-                if update.id in zino_cases:
-                    del zino_cases[update.id]
-
-            elif old_state == "embryonic" and new_state == "open":
-                # Newly created case
-                case = _zino.case(update.id)
-                if update.id not in argus_incidents:
-                    if not is_case_interesting(case):
-                        continue
-                    _logger.debug(
-                        "Creating zino case %s as incident in argus", update.id
-                    )
-                    zino_cases[update.id] = case
-                    argus_incidents[update.id] = create_argus_incident(case)
-                else:
-                    _logger.debug("Zino case {} is already added to argus")
-            else:
-                # All other state changes
-                # zino_cases[update.id] = _zino.case(update.id)
-                pass
-        if update.type == "log":
-            _logger.debug(
-                "Log message received for %s checking if case is in argus", update.id
-            )
-            case = _zino.case(update.id)
-            if update.id not in argus_incidents:
-                # Create ticket if we care about it
-                if not is_case_interesting(case):
-                    continue
-                zino_cases[update.id] = case
-                argus_incidents[update.id] = create_argus_incident(case)
```

Comment:
Has this more or less been moved to before `if update.type == "state":`? Because I can't find  `if update.type == "log":` anywhere.

Reply:
This has been removed.  If you look more closely, it doesn't actually do anything log-entry specific.  All it does is repeat the same code that creates an Argus incident if one doesn't exist - the same code contained in all of the original if statement branches.

This silly code branch is more or less the reason I set out to make this PR :laughing: